codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/rustlang/WritableTest.kt

```diff
@@ -75,4 +75,36 @@ internal class RustTypeParametersTest {
 
         writer.toString() shouldContain "'<()>'"
     }
+
+    @Test
+    fun `join iterable`() {
+        val writer = RustWriter.forModule("model")
+        listOf(writable("A"), writable("B"), writable("C")).join("-")(writer)
+        listOf(writable("D"), writable("E"), writable("F")).join(writable("+"))(writer)
+        writer.toString() shouldContain "A-B-CD+E+F"
```

Comment:
I'd probably make this test a bit more realistic, eg. make a `vec![...]` out of a series of writeables.